subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6711,6 +6719,24 @@ static void le_per_adv_sync_report(struct pdu_data *pdu_data,
 	if ((le_event_mask & BT_EVT_MASK_LE_PER_ADVERTISING_REPORT) && accept &&
 	    ((data_len_total - data_len) < CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX)) {
 
+		/* Pass verdict in LL.TS.p19 section 4.2.3.6 Extended Scanning,
+		 * Passive, Periodic Advertising Report, RSSI and TX_Power
+		 * states:
+		 * TX_Power is set to value of the TxPower field for the
+		 * AUX_SYNC_IND received, and RSSI set to a valid value.
+		 * Subsequent reports with data and the status set to
+		 * "Incomplete, more data to come" or "complete" can have the
+		 * TX_Power field set to 0x7F.
+		 *
+		 * In the implementation data_len_total is the running total
+		 * AD data length so far, data_len is the current PDU's AD data
+		 * length.
+		 */
+		if (data_len_total > data_len) {
+			/* Subsequent report */
+			tx_pwr = BT_HCI_LE_ADV_TX_POWER_NO_PREF;
+		}
```

Comment:
It's better, but it still isn't clear (to me) **why** if `data_len_total > data_len` then `tx_pwr = BT_HCI_LE_ADV_TX_POWER_NO_PREF;`.

Reply:
if `data_len_total > data_len` then current PDU is generating a subsequent report i.e. after the first "incomplete, more data to come".  First PDU (i.e. AUX_SYNC_IND) would have `data_len_total == data_len`.

Reply:
Alright, thanks :) Can you add that as part of the comment there? 

Is the `tx_pwr` then only transmitted in the last (or complete) PDU?

Reply:
A valid Tx_pwr is present in the first incomplete report or first complete report.